ui/app/AppLayouts/Profile/helpers/VisibilityAndPositionDirtyStateModel.qml

```diff
@@ -0,0 +1,106 @@
+import QtQml 2.15
+
+import StatusQ 0.1
+import StatusQ.Core.Utils 0.1
+
+/**
+  * Basic building block for storing temporary state in the "Profile Showcase"
+  * functionality. Allows to store on the UI side the temporary position and
+  * visibility level. Can store temporary state for Communities, Accounts,
+  * Collectibles and Assets.
+  */
+WritableProxyModel {
+    id: root
+
+    /* Provides the list of objects representing the current state in the
+     * in the following format:
+     * [ {
+     *     key: <string or integer>
+     *     position: <integer>
+     *     visibility: <integer>
+     *   }
+     * ]
+     *
+     * The enteries with visibility 0 (hidden) are not included in the list.
+     */
+    function currentState() {
+        const visible = d.getVisibleEntries()
+        const minPos = Math.min(...visible.map(e => e.position))
+
+        return visible.map(e => { e.position -= minPos; return e })
+    }
+
+    /* Sets the visibility of the given item. If the element was hidden, it is
+     * positioned last.
+     */
+    function setVisibility(key, visibility) {
+        const sourceIdx = d.indexByKey(key)
+        const oldVisibility = d.getVisibility(sourceIdx)
+
+        if (oldVisibility === visibility)
+            return
+
+        // hiding, changing visibility level
+        if (visibility === 0 || (visibility > 0 && oldVisibility > 0)) {
```

Comment:
Yes, I started from number like 0 (hidden) and 1, 2, 3 for visibility levers and finally left it in this form. It should be excluded to separate enum or constants for sure, but maybe we could handle this part in integration phase to cover backend, ui and this part consistently.

Reply:
Yeah I agree with this point, and since the component is inside the `Profile/helpers/`, I'd directly use the existing enum from `Constants.qml` for this purpose:
```qml
enum ShowcaseVisibility {
    NoOne = 0,
    IdVerifiedContacts = 1,
    Contacts = 2,
    Everyone = 3
}
```

Reply:
Can we create a task for not forgetting this point?

Reply:
Done :)